src/controllers/expenses.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const userService = require('../services/users.js');
+const expenseService = require('../services/expenses.js');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    from,
+    to,
+    category,
+  } = req.query;
+
+  let expenses = expenseService.getAll();
+
+  if (userId) {
+    expenses = expenseService.filterAllByUserId(userId);
+  }
+
+  if (from && to) {
+    expenses = expenseService.filterAllByDate(from, to);
+  }
+
+  if (category) {
+    expenses = expenseService.filterAllByCategory(category);
+  }
```

Comment:
nice solution with separete functions here

Reply:
Thanks for reviewing 😃